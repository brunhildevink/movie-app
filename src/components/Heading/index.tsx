import styled from 'styled-components'

import { fonts, typography, breakpoints } from '../../styles'

const sharedCss = `
  font-family: ${fonts};

  &:first-child {
    margin-top: 0;
  }

  &:last-child {
    margin-bottom: 0;
  }
`

const HeadingOne = styled.h1`
  font-size: ${typography.fontSizeHeadingOneMobile}px;
  line-height: ${typography.lineHeightHeadingOneMobile};
  font-weight: ${typography.fontWeightExtraBold};

  @media ${breakpoints.md} {
    font-size: ${typography.fontSizeHeadingOne}px;
    line-height: ${typography.lineHeightHeadingOne};
  }

  ${sharedCss}
`

const HeadingTwo = styled.h2`
  font-size: ${typography.fontSizeHeadingTwoMobile}px;
  line-height: ${typography.lineHeightHeadingTwoMobile};
  font-weight: ${typography.fontWeightBold};

  @media ${breakpoints.md} {
    font-size: ${typography.fontSizeHeadingTwo}px;
    line-height: ${typography.lineHeightHeadingTwo};
  }

  ${sharedCss}
`

const HeadingThree = styled.h3`
  font-size: ${typography.fontSizeHeadingThreeMobile}px;
  line-height: ${typography.lineHeightHeadingThreeMobile};
  font-weight: ${typography.fontWeightBold};

  @media ${breakpoints.md} {
    font-size: ${typography.fontSizeHeadingThree}px;
    line-height: ${typography.lineHeightHeadingThree};
  }

  ${sharedCss}
`

const HeadingFour = styled.h4`
  font-size: ${typography.fontSizeHeadingFourMobile}px;
  line-height: ${typography.lineHeightHeadingFourMobile};
  font-weight: ${typography.fontWeightBold};

  @media ${breakpoints.md} {
    font-size: ${typography.fontSizeHeadingFour}px;
    line-height: ${typography.lineHeightHeadingFour};
  }

  ${sharedCss}
`

export default { HeadingOne, HeadingTwo, HeadingThree, HeadingFour }
