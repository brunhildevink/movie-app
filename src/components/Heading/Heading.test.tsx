import React from 'react'
import renderer from 'react-test-renderer'
import Heading from '.'

const { HeadingOne, HeadingTwo, HeadingThree, HeadingFour } = Heading

test('Heading one renders correctly', () => {
  const tree = renderer.create(<HeadingOne />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading two renders correctly', () => {
  const tree = renderer.create(<HeadingTwo />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading three renders correctly', () => {
  const tree = renderer.create(<HeadingThree />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading three renders correctly', () => {
  const tree = renderer.create(<HeadingFour />).toJSON()
  expect(tree).toMatchSnapshot()
})
