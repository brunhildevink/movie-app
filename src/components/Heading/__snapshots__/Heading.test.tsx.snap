// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading one renders correctly 1`] = `
<h1
  className="sc-bczRLJ iluBtY"
/>
`;

exports[`Heading three renders correctly 1`] = `
<h3
  className="sc-dkzDqf dsgIKT"
/>
`;

exports[`Heading three renders correctly 2`] = `
<h4
  className="sc-hKMtZM gkfzAF"
/>
`;

exports[`Heading two renders correctly 1`] = `
<h2
  className="sc-gsnTZi culZzQ"
/>
`;
